src/commands/init/index.ts

```diff
@@ -47,26 +58,78 @@ export default class Init extends Command {
-        const projectConfig = this.initializeOrReadProject();
+        const projectConfig = this._initializeOrReadProject();
 
         const appManifestData = AppManifest.read(projectConfig.getVariableMappings().get(APP_MANIFEST_PATH_VARIABLE));
 
-        await this.ensurePackagesAreDownloaded(projectConfig, flags.force, flags.verbose);
-        projectConfig.validateUsedComponents();
+        let packageConfigs = projectConfig.getPackages(false);
+        if (flags.package !== '') {
+            const requestedPackageConfig = this._parsePackageConfig(flags.package);
+            await this._resolveVersion(requestedPackageConfig, flags.verbose);
+            await this._modifyProjectConfig(requestedPackageConfig, projectConfig, flags.force, flags.verbose);
+            await this._ensurePackageIsDownloaded(requestedPackageConfig, flags.force, flags.verbose);
+            this._addComponentsToProjectConfig(requestedPackageConfig, projectConfig);
+            // projectConfig.validateUsedComponents();
+        } else {
+            for (const packageConfig of packageConfigs) {
+                await this._resolveVersion(packageConfig, flags.verbose);
+                await this._ensurePackageIsDownloaded(packageConfig, flags.force, flags.verbose);
+            }
+            projectConfig.validateUsedComponents();
+        }
 
         if (!flags['no-hooks']) {
-            await this.runPostInitHooks(projectConfig, appManifestData, flags.verbose);
+            await this._runPostInitHooks(projectConfig, appManifestData, flags.verbose);
         }
 
-        this.createProjectLockFile(projectConfig, flags.verbose);
+        projectConfig.write();
+        this._createProjectLockFile(projectConfig, flags.verbose);
     }
 
-    initializeOrReadProject(): ProjectConfig {
+    private _addComponentsToProjectConfig(requestedPackageConfig: PackageConfig, projectConfig: ProjectConfig) {
+        if (requestedPackageConfig !== null) {
+            const providedComponents = requestedPackageConfig.readPackageManifest().components;
+            const areComponentsExisting = providedComponents.some((comp) => {
+                const enabledComponents = projectConfig.getComponentsFromInstalledPackages().map((comp) => comp.config.id);
+                return enabledComponents.indexOf(comp.id) > -1;
+            });
+
+            if (!areComponentsExisting) {
+                providedComponents.forEach((providedComponent) => {
+                    projectConfig.addComponent(providedComponent.id);
+                });
+            }
+        }
+    }
+
+    private _parsePackageConfig(packageFlag: string): PackageConfig {
+        const hasRequestedVersion = packageFlag.indexOf('@') > -1;
+
+        let repo: string;
+        let requestedVersion: string;
+        if (hasRequestedVersion) {
+            repo = packageFlag.substring(0, packageFlag.indexOf('@'));
+            requestedVersion = packageFlag.substring(packageFlag.indexOf('@') + 1);
```

Comment:
this is not working when we e.g call:

velocitas init -p git@github.com:eclipse-velocitas/devenv-runtimes.git@v3

Reply:
Good find. Simply replacing indexOf by lastIndexOf will lead to issues when urls like this without version (e.g.: git@github.com:eclipse-velocitas/devenv-runtimes.git) are provided.

Maybe we should introduce another optional flag '--version' to specify the version? Do you have a better / easier idea?

Reply:
```
    private _parsePackageConfig(packageFlag: string): PackageConfig {
        let packageName: string, packageVersion: string;
        if (packageFlag.startsWith('git@')) {
            const [ignored, restOfPackage] = packageFlag.split('git@');
            [packageName, packageVersion = ''] = restOfPackage.split('@');
            packageName = `git@${packageName}`;
        } else {
            [packageName, packageVersion = ''] = packageFlag.split('@');
        }

        const pkgCfg = new PackageConfig({ repo: packageName, version: packageVersion });
        await this._resolveVersion(pkgCfg, flags.verbose);
        return pkgCfg;
    }
```

Reply:
Good solution, I still see two downsides: 
1. It most probably works only for default repo providers, because git is a username and therefore different providers might require the use of different usernames over the git one
2. for https a valid format would be the following: https://username:password@github.com/username/repository.git - I could still imagine people with private repos checking in username:password combinations here

We can still accept this behavior, but we probably should put it somewhere as a "known issue", to demonstrate which urls are / are not supported.